### PR TITLE
Create branch during cherry-picking via the context menu

### PR DESCRIPTION
**Pull request by tidy-dev**
Adding to #1685

## Description

Add ability to create a branch during cherry-picking via the context menu.

### Screenshots

https://user-images.githubusercontent.com/75402236/113047818-c9532980-916f-11eb-880d-6c64467e68c7.mp4

## Release notes

Notes: [Added]  User can create a branch during cherry-picking via the context menu.

### CODE REVIEW

**Review by billygriffin — DISMISSED** [review]
From a product perspective, I think this is just 💯  Great work!

File: app/src/lib/stores/app-store.ts [hunk 1]
```diff
@@ -3009,14 +3009,17 @@ export class AppStore extends TypedBaseStore<IAppState> {
     repository: Repository,
     name: string,
     startPoint: string | null,
-    noTrackOption: boolean = false
-  ): Promise<void> {
+    noTrackOption: boolean = false,
+    checkoutBranch: boolean = false
```
Comment (sergiou87) [inline app/src/lib/stores/app-store.ts]: I couldn't see where this new parameter is used. Is this a leftover?

File: app/src/lib/stores/app-store.ts [hunk 2]
```diff
@@ -3442,7 +3446,8 @@ export class AppStore extends TypedBaseStore<IAppState> {
 
       // If a local branch, user may have the branch to delete checked out and
       // we need to switch to a different branch (default or recent).
-      const branchToCheckout = this.getBranchToCheckoutAfterDelete(branch, r)
+      const branchToCheckout =
+        toCheckout || this.getBranchToCheckoutAfterDelete(branch, r)
```
Comment (sergiou87) [inline app/src/lib/stores/app-store.ts]: Nit: Should we use `??` here? I understand they should be equivalent in this specific case, but `||` does a falsy/truthy check which, in theory, we should avoid 🤔 
```suggestion
        toCheckout ?? this.getBranchToCheckoutAfterDelete(branch, r)
```

**Review by sergiou87 — CHANGES_REQUESTED** [review]
Only found a couple of minor issues. Other than that it looks good and works beautifully well!! 😍

File: app/src/lib/stores/app-store.ts (hunk 2 above)
Comment (tidy-dev) [reply app/src/lib/stores/app-store.ts]: Had to look that one up I had always used/seen `||` before - and yeah `??` makes more sense.

File: app/src/lib/stores/app-store.ts (hunk 1 above)
Comment (tidy-dev) [reply app/src/lib/stores/app-store.ts]: Ugg.. good catch.. that should default to true..

Comment (tidy-dev) [reply app/src/lib/stores/app-store.ts]: And start cherry-pick with target branch name in dispatcher should set to false.  (in other words this was a regression)

**Review by sergiou87 — APPROVED** [review]
LGTM now!! Thanks for those changes 😄  :shipit: 🚀

**Review by sirpenev — COMMENTED** [review]
T

**Pull request merged**